### DISCUSSION TITLE
[Feature Request] Support for nested tables

### DISCUSSION THREAD

**Discussion by Marko-D** [heroui-inc/heroui#4734] [Feature requests]
### Is your feature request related to a problem? Please describe.

The problem is that currently I cannot nest table inside each row.  
I've tried something like this

![image](https://github.com/user-attachments/assets/a542b70f-9e87-4b1d-a55e-24b485d2022b)
TypeError: Cannot convert a Symbol value to a string

### Describe the solution you'd like

As a user I expect to be able to add whathever component I need inside TableRow and render it next to the row

### Describe alternatives you've considered

To be able to pass custom component where the row is renderd

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1369/[feature-request]-support-for-nested-tables">ENG-1369 [Feature Request] Support for nested tables</a></p>

**Comment by buchananwill** [comment]
Do you need to add the 'use client'  directive at the top of your file? This error has throw me many times.

If not that, have you tried using the items prop on the table body?

```
 <TableBody emptyContent={'No content found'} items={visibleItems}>
        {(item) => (
          <TableRow key={item.id}>
            {(columnKey) => (
              <TableCell>{renderCell(item, columnKey)}</TableCell>
            )}
          </TableRow>
        )}
      </TableBody>
```

**Comment by Marko-D** [comment]
I have the 'use client' directive added. Even when using the items prop i het the same error: TypeError: Cannot convert a Symbol value to a string. 

A similar issue has been reported here https://github.com/nextui-org/nextui/issues/2564 but my use case is I want to render a nested table for each row that will be togglable, but this is for nesting inside a column which is not what I need.
